torch/_library/autograd.py

```diff
@@ -48,7 +55,13 @@ def backward(ctx, *grads):
     ):
         Generated = supports_tensorlist(Generated)
 
-    def autograd_impl(*args):
+    def autograd_impl(keyset, *args):
+        # We set a nonlocal to ferry keyset from here to the forward.
```

Comment:
Why not pass it as an argument?

Reply:
this changes `ctx.needs_input_grad` and I don't want to fix that right now haha